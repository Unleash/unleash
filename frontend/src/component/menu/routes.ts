import { FeatureToggleListTable } from 'component/feature/FeatureToggleList/FeatureToggleListTable';
import { StrategyView } from 'component/strategies/StrategyView/StrategyView';
import { StrategiesList } from 'component/strategies/StrategiesList/StrategiesList';
import { TagTypeList } from 'component/tags/TagTypeList/TagTypeList';
import { AddonList } from 'component/addons/AddonList/AddonList';
import Admin from 'component/admin';
import AdminApi from 'component/admin/api';
import AdminUsers from 'component/admin/users/UsersAdmin';
import { GroupsAdmin } from 'component/admin/groups/GroupsAdmin';
import { AuthSettings } from 'component/admin/auth/AuthSettings';
import Login from 'component/user/Login/Login';
import { EEA, P, RE, SE, UG } from 'component/common/flags';
import { NewUser } from 'component/user/NewUser/NewUser';
import ResetPassword from 'component/user/ResetPassword/ResetPassword';
import ForgottenPassword from 'component/user/ForgottenPassword/ForgottenPassword';
import { ProjectListNew } from 'component/project/ProjectList/ProjectList';
import Project from 'component/project/Project/Project';
import RedirectArchive from 'component/archive/RedirectArchive';
import { FeatureView } from 'component/feature/FeatureView/FeatureView';
import ProjectRoles from 'component/admin/projectRoles/ProjectRoles/ProjectRoles';
import CreateProjectRole from 'component/admin/projectRoles/CreateProjectRole/CreateProjectRole';
import EditProjectRole from 'component/admin/projectRoles/EditProjectRole/EditProjectRole';
import CreateUser from 'component/admin/users/CreateUser/CreateUser';
import EditUser from 'component/admin/users/EditUser/EditUser';
import { CreateApiToken } from 'component/admin/apiToken/CreateApiToken/CreateApiToken';
import CreateEnvironment from 'component/environments/CreateEnvironment/CreateEnvironment';
import EditEnvironment from 'component/environments/EditEnvironment/EditEnvironment';
import { EditContext } from 'component/context/EditContext/EditContext';
import EditTagType from 'component/tags/EditTagType/EditTagType';
import CreateTagType from 'component/tags/CreateTagType/CreateTagType';
import EditProject from 'component/project/Project/EditProject/EditProject';
import CreateProject from 'component/project/Project/CreateProject/CreateProject';
import CreateFeature from 'component/feature/CreateFeature/CreateFeature';
import EditFeature from 'component/feature/EditFeature/EditFeature';
import { ApplicationEdit } from 'component/application/ApplicationEdit/ApplicationEdit';
import { ApplicationList } from 'component/application/ApplicationList/ApplicationList';
import ContextList from 'component/context/ContextList/ContextList';
import RedirectFeatureView from 'component/feature/RedirectFeatureView/RedirectFeatureView';
import { CreateAddon } from 'component/addons/CreateAddon/CreateAddon';
import { EditAddon } from 'component/addons/EditAddon/EditAddon';
import { CopyFeatureToggle } from 'component/feature/CopyFeature/CopyFeature';
import { EventPage } from 'component/events/EventPage/EventPage';
import { CreateStrategy } from 'component/strategies/CreateStrategy/CreateStrategy';
import { EditStrategy } from 'component/strategies/EditStrategy/EditStrategy';
import { SplashPage } from 'component/splash/SplashPage/SplashPage';
import { CreateUnleashContextPage } from 'component/context/CreateUnleashContext/CreateUnleashContextPage';
import { CreateSegment } from 'component/segments/CreateSegment/CreateSegment';
import { EditSegment } from 'component/segments/EditSegment/EditSegment';
import { IRoute } from 'interfaces/route';
import { EnvironmentTable } from 'component/environments/EnvironmentTable/EnvironmentTable';
import { SegmentTable } from 'component/segments/SegmentTable/SegmentTable';
import FlaggedBillingRedirect from 'component/admin/billing/FlaggedBillingRedirect/FlaggedBillingRedirect';
import { FeaturesArchiveTable } from '../archive/FeaturesArchiveTable';
import { Billing } from 'component/admin/billing/Billing';
import { Group } from 'component/admin/groups/Group/Group';
import { CreateGroup } from 'component/admin/groups/CreateGroup/CreateGroup';
import { EditGroup } from 'component/admin/groups/EditGroup/EditGroup';
import { LazyPlayground } from 'component/playground/Playground/LazyPlayground';
import { CorsAdmin } from 'component/admin/cors';
import { InviteLink } from 'component/admin/users/InviteLink/InviteLink';
import { Profile } from 'component/user/Profile/Profile';

export const routes: IRoute[] = [
    // Splash
    {
        path: '/splash/:splashId',
        title: 'Unleash',
        component: SplashPage,
        type: 'protected',
        menu: {},
    },

    // Project
    {
        path: '/projects/create',
        parent: '/projects',
        title: 'Create',
        component: CreateProject,
        type: 'protected',
        enterprise: true,
        menu: {},
    },
    {
        path: '/projects/:projectId/edit',
        parent: '/projects',
        title: ':projectId',
        component: EditProject,
        type: 'protected',
        enterprise: true,
        menu: {},
    },
    {
        path: '/projects/:projectId/archived',
        title: ':projectId',
        parent: '/archive',
        component: RedirectArchive,
        type: 'protected',
        menu: {},
    },
    {
        path: '/projects/:projectId/features/:featureId/:activeTab/copy',
        parent: '/projects/:projectId/features/:featureId/:activeTab',
        title: 'Copy',
        component: CopyFeatureToggle,
        type: 'protected',
        menu: {},
    },
    {
        path: '/projects/:projectId/features/:featureId/edit',
        parent: '/projects',
        title: 'Edit feature',
        component: EditFeature,
        type: 'protected',
        menu: {},
    },
    {
        path: '/projects/:projectId/features/:featureId/*',
        parent: '/projects',
        title: 'FeatureView',
        component: FeatureView,
        type: 'protected',
        menu: {},
    },
    {
        path: '/projects/:projectId/create-toggle',
        parent: '/projects/:projectId/features',
        title: 'Create feature toggle',
        component: CreateFeature,
        type: 'protected',
        menu: {},
    },
    {
        path: '/projects/:projectId/features2/:featureId',
        parent: '/features',
        title: ':featureId',
        component: RedirectFeatureView,
        type: 'protected',
        menu: {},
    },
    {
        path: '/projects/:projectId/*',
        parent: '/projects',
        title: ':projectId',
        component: Project,
        flag: P,
        type: 'protected',
        menu: {},
    },
    {
        path: '/projects',
        title: 'Projects',
        component: ProjectListNew,
        type: 'protected',
        menu: { mobile: true },
    },

    // Features
    {
        path: '/features/:activeTab/:featureId',
        parent: '/features',
        title: ':featureId',
        component: RedirectFeatureView,
        type: 'protected',
        menu: {},
    },
    {
        path: '/features',
        title: 'Feature toggles',
        component: FeatureToggleListTable,
        type: 'protected',
        menu: { mobile: true },
    },

    // Playground
    {
        path: '/playground',
        title: 'Playground',
        component: LazyPlayground,
        hidden: false,
        type: 'protected',
        menu: { mobile: true },
    },

    // Applications
    {
        path: '/applications/:name',
        title: ':name',
        parent: '/applications',
        component: ApplicationEdit,
        type: 'protected',
        menu: {},
    },
    {
        path: '/applications',
        title: 'Applications',
        component: ApplicationList,
        type: 'protected',
        menu: { mobile: true, advanced: true },
    },

    // Context
    {
        path: '/context/create',
        parent: '/context',
        title: 'Create',
        component: CreateUnleashContextPage,
        type: 'protected',
        menu: {},
    },
    {
        path: '/context/edit/:name',
        parent: '/context',
        title: ':name',
        component: EditContext,
        type: 'protected',
        menu: {},
    },
    {
        path: '/context',
        title: 'Context fields',
        component: ContextList,
        type: 'protected',
        menu: { mobile: true, advanced: true },
    },

    // Strategies
    {
        path: '/strategies/create',
        title: 'Create',
        parent: '/strategies',
        component: CreateStrategy,
        type: 'protected',
        menu: {},
    },
    {
        path: '/strategies/:name/edit',
        title: ':name',
        parent: '/strategies',
        component: EditStrategy,
        type: 'protected',
        menu: {},
    },
    {
        path: '/strategies/:name',
        title: ':name',
        parent: '/strategies',
        component: StrategyView,
        type: 'protected',
        menu: {},
    },
    {
        path: '/strategies',
        title: 'Strategy types',
        component: StrategiesList,
        type: 'protected',
        menu: { mobile: true, advanced: true },
    },
    {
        path: '/environments/create',
        title: 'Environments',
        component: CreateEnvironment,
        parent: '/environments',
        type: 'protected',
        menu: {},
    },
    {
        path: '/environments/:id',
        title: 'Edit',
        component: EditEnvironment,
        type: 'protected',
        menu: {},
    },
    {
        path: '/environments',
        title: 'Environments',
        component: EnvironmentTable,
        type: 'protected',
        flag: EEA,
        menu: { mobile: true, advanced: true },
    },

    // Tags
    {
        path: '/tag-types/create',
        parent: '/tag-types',
        title: 'Create',
        component: CreateTagType,
        type: 'protected',
        menu: {},
    },
    {
        path: '/tag-types/edit/:name',
        parent: '/tag-types',
        title: ':name',
        component: EditTagType,
        type: 'protected',
        menu: {},
    },
    {
        path: '/tag-types',
        title: 'Tag types',
        component: TagTypeList,
        type: 'protected',
        menu: { mobile: true, advanced: true },
    },

    // Addons
    {
        path: '/addons/create/:providerId',
        parent: '/addons',
        title: 'Create',
        component: CreateAddon,
        type: 'protected',
        menu: {},
    },
    {
        path: '/addons/edit/:addonId',
        parent: '/addons',
        title: 'Edit',
        component: EditAddon,
        type: 'protected',
        menu: {},
    },
    {
        path: '/addons',
        title: 'Addons',
        component: AddonList,
        hidden: false,
        type: 'protected',
        menu: { mobile: true, advanced: true },
    },

    // Segments
    {
        path: '/segments/create',
        title: 'Segments',
        component: CreateSegment,
        hidden: false,
        type: 'protected',
        layout: 'main',
        menu: {},
        flag: SE,
    },
    {
        path: '/segments/edit/:segmentId',
        title: 'Segments',
        component: EditSegment,
        hidden: false,
        type: 'protected',
        layout: 'main',
        menu: {},
        flag: SE,
    },
    {
        path: '/segments',
        title: 'Segments',
        component: SegmentTable,
        hidden: false,
        type: 'protected',
        menu: { mobile: true, advanced: true },
        flag: SE,
    },

    // History
    {
        path: '/history',
        title: 'Event log',
        component: EventPage,
        type: 'protected',
        menu: { adminSettings: true },
    },

    // Archive
    {
        path: '/archive',
        title: 'Archived toggles',
        component: FeaturesArchiveTable,
        type: 'protected',
        menu: {},
    },

    // Admin
    {
        path: '/admin/api/create-token',
        parent: '/admin',
        title: 'API access',
        component: CreateApiToken,
        type: 'protected',
        menu: {},
    },
    {
        path: '/admin/create-project-role',
        title: 'Create',
        component: CreateProjectRole,
        type: 'protected',
        menu: {},
        flag: RE,
    },
    {
        path: '/admin/roles/:id/edit',
        title: 'Edit',
        component: EditProjectRole,
        type: 'protected',
        menu: {},
        flag: RE,
    },
    {
        path: '/admin/users/:id/edit',
        title: 'Edit',
        component: EditUser,
        type: 'protected',
        menu: {},
        hidden: true,
    },
    {
        path: '/admin/api',
        parent: '/admin',
        title: 'API access',
        component: AdminApi,
        type: 'protected',
        menu: { mobile: true, advanced: true },
    },
    {
        path: '/admin/users',
        parent: '/admin',
        title: 'Users',
        component: AdminUsers,
        type: 'protected',
        menu: { adminSettings: true },
    },
    {
        path: '/admin/create-user',
        parent: '/admin',
        title: 'Users',
        component: CreateUser,
        type: 'protected',
        menu: {},
    },
    {
        path: '/admin/invite-link',
        parent: '/admin',
        title: 'Invite link',
        component: InviteLink,
        type: 'protected',
        menu: {},
    },
    {
        path: '/admin/groups',
        parent: '/admin',
        title: 'Groups',
        component: GroupsAdmin,
        type: 'protected',
        menu: { adminSettings: true },
        flag: UG,
    },
    {
        path: '/admin/groups/:groupId',
        parent: '/admin',
        title: ':groupId',
        component: Group,
        type: 'protected',
        menu: {},
        flag: UG,
    },
    {
        path: '/admin/groups/create-group',
        parent: '/admin/groups',
        title: 'Create group',
        component: CreateGroup,
        type: 'protected',
        menu: {},
        flag: UG,
    },
    {
        path: '/admin/groups/:groupId/edit',
        parent: '/admin/groups',
        title: 'Edit group',
        component: EditGroup,
        type: 'protected',
        menu: {},
        flag: UG,
    },
    {
        path: '/admin/roles',
        parent: '/admin',
        title: 'Project roles',
        component: ProjectRoles,
        type: 'protected',
        flag: RE,
        menu: { adminSettings: true },
    },
    {
        path: '/admin/auth',
        parent: '/admin',
        title: 'Single sign-on',
        component: AuthSettings,
        type: 'protected',
        menu: { adminSettings: true },
    },
    {
        path: '/admin/cors',
        parent: '/admin',
        title: 'CORS origins',
        component: CorsAdmin,
        type: 'protected',
        flag: 'embedProxyFrontend',
        menu: { adminSettings: true },
    },
    {
        path: '/admin/billing',
        parent: '/admin',
        title: 'Billing',
        component: Billing,
        type: 'protected',
        menu: {},
    },
    {
        path: '/admin-invoices',
        parent: '/admin',
        title: 'Billing & invoices',
        component: FlaggedBillingRedirect,
        type: 'protected',
        menu: { adminSettings: true, isEnterprise: true },
    },
    {
        path: '/admin',
        title: 'Admin',
        component: Admin,
        hidden: false,
        type: 'protected',
        menu: {},
    },
    {
        path: '/profile/*',
        title: 'Profile',
        component: Profile,
        type: 'protected',
        menu: {},
    },

    /* If you update this route path, make sure you update the path in SWRProvider.tsx */
    {
        path: '/login',
        title: 'Log in',
        component: Login,
        type: 'unprotected',
        hidden: true,
        menu: {},
    },
    /* If you update this route path, make sure you update the path in SWRProvider.tsx */
    {
        path: '/new-user',
        title: 'New user',
        hidden: true,
        component: NewUser,
        type: 'unprotected',
        menu: {},
    },
    /* If you update this route path, make sure you update the path in SWRProvider.tsx */
    {
        path: '/reset-password',
        title: 'Reset password',
        hidden: true,
        component: ResetPassword,
        type: 'unprotected',
        menu: {},
    },
    /* If you update this route path, make sure you update the path in SWRProvider.tsx */
    {
        path: '/forgotten-password',
        title: 'Forgotten password',
        hidden: true,
        component: ForgottenPassword,
        type: 'unprotected',
        menu: {},
    },
];

export const getRoute = (path: string) =>
    routes.find(route => route.path === path);

export const baseRoutes = routes.filter(route => !route.hidden);

const computeRoutes = () => {
    const mainNavRoutes = baseRoutes.filter(route => route.menu.advanced);
    const adminRoutes = routes.filter(route => route.menu.adminSettings);
    const mobileRoutes = routes.filter(route => route.menu.mobile);

    const computedRoutes = {
        mainNavRoutes,
        adminRoutes,
        mobileRoutes,
    };
    return () => {
        return computedRoutes;
    };
};

export const getRoutes = computeRoutes();
