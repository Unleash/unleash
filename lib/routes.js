var knex = require('./dbPool');
var logger = require('./logger');

module.exports = function (app) {
    app.get('/health', function (req, res) {
        knex.select(1)
            .from('features')
            .then(function() {
                res.json({health: 'GOOD'});
            })
            .catch(function(err) {
                logger.error('Could not select from features, error was: ', err);
                res.status(500).json({health: 'BAD'});
            });
    });
};
