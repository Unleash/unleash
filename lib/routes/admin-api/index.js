'use strict';

const Controller = require('../controller');
const FeatureController = require('./feature.js');
const FeatureTypeController = require('./feature-type.js');
const ArchiveController = require('./archive.js');
const EventController = require('./event.js');
const StrategyController = require('./strategy');
const MetricsController = require('./metrics');
const UserController = require('./user');
const ConfigController = require('./config');
const ContextController = require('./context');
const StateController = require('./state');
const apiDef = require('./api-def.json');

class AdminApi extends Controller {
    constructor(config) {
        super(config);

        this.app.get('/', this.index);
        this.app.use('/features', new FeatureController(config).router);
        this.app.use(
            '/feature-types',
            new FeatureTypeController(config).router,
        );
        this.app.use('/archive', new ArchiveController(config).router);
        this.app.use('/strategies', new StrategyController(config).router);
        this.app.use('/events', new EventController(config).router);
        this.app.use('/metrics', new MetricsController(config).router);
        this.app.use('/user', new UserController(config).router);
        this.app.use('/ui-config', new ConfigController(config).router);
        this.app.use('/context', new ContextController(config).router);
        this.app.use('/state', new StateController(config).router);
    }

    index(req, res) {
        res.json(apiDef);
    }
}

module.exports = AdminApi;
