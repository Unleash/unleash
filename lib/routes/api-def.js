'use strict';

// export module version
require('pkginfo')(module, 'version');

const { version } = module.exports;
const clientApiDef = require('./client-api/api-def.json');
const adminApiDef = require('./admin-api/api-def.json');

const apiDef = {
    name: 'unleash-server',
    version,
    uri: '/api',
    links: {
        admin: {
            uri: '/api/admin',
            links: adminApiDef.links,
        },
        client: {
            uri: '/api/client',
            links: clientApiDef.links,
        },
    },
};

module.exports = apiDef;
