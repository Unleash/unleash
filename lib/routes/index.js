'use strict';

const { Router } = require('express');
const AdminApi = require('./admin-api');
const ClientApi = require('./client-api');
const FeatureController = require('./client-api/feature.js');

const HealthCheckController = require('./health-check');
const BackstageCTR = require('./backstage.js');
const api = require('./api-def');

class IndexRouter {
    constructor(config) {
        const router = Router();
        this._router = router;

        router.use('/health', new HealthCheckController(config).router());
        router.use('/internal-backstage', new BackstageCTR(config).router());
        router.get(api.uri, (req, res) => res.json(api));
        router.use(api.links.admin.uri, new AdminApi(config).router());
        router.use(api.links.client.uri, new ClientApi(config).router());

        // legacy support (remove in 4.x)
        if (config.enableLegacyRoutes) {
            router.use(
                '/api/features',
                new FeatureController(config.stores).router()
            );
        }
    }

    router() {
        return this._router;
    }
}

module.exports = IndexRouter;
