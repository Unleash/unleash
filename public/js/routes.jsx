import React from 'react'
import { DefaultRoute, Router, Route} from 'react-router'
import UnleashApp from './UnleashApp'
import LogEntriesComponent from './components/log/LogEntriesComponent'
import FeatureTogglesComponent from './components/feature/FeatureTogglesComponent'
import StrategiesComponent from './components/strategy/StrategiesComponent'
import ArchiveFeatureComponent from './components/feature/ArchiveFeatureComponent'

const routes = (
    <Route name="app" path="/" handler={UnleashApp}>
        <Route name="strategies" handler={StrategiesComponent}/>
        <Route name="log" handler={LogEntriesComponent}/>
        <Route name="archive" handler={ArchiveFeatureComponent}/>
        <DefaultRoute name="features" handler={FeatureTogglesComponent}/>
    </Route>
);

module.exports = routes;
