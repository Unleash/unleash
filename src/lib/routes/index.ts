import { BackstageController } from './backstage';
import ResetPasswordController from './auth/reset-password-controller';
import { SimplePasswordProvider } from './auth/simple-password-provider';
import { IUnleashConfig, IUnleashServices } from '../types';
import LogoutController from './logout';
import rateLimit from 'express-rate-limit';

const AdminApi = require('./admin-api');
const ClientApi = require('./client-api');
const Controller = require('./controller');
import { HealthCheckController } from './health-check';
import ProxyController from './proxy-api';
import { conditionalMiddleware } from '../middleware/conditional-middleware';
import EdgeController from './edge-api';
import { PublicInviteController } from './public-invite';

class IndexRouter extends Controller {
    constructor(config: IUnleashConfig, services: IUnleashServices) {
        super(config);

        this.use('/health', new HealthCheckController(config, services).router);
        this.use(
            '/invite',
            new PublicInviteController(config, services).router,
        );
        this.use('/internal-backstage', new BackstageController(config).router);
        this.use('/logout', new LogoutController(config, services).router);
        this.useWithMiddleware(
            '/auth/simple',
            new SimplePasswordProvider(config, services).router,
            rateLimit({
                windowMs: 1 * 60 * 1000,
                max: 5,
                standardHeaders: true,
                legacyHeaders: false,
            }),
        );
        this.use(
            '/auth/reset',
            new ResetPasswordController(config, services).router,
        );

        this.use('/api/admin', new AdminApi(config, services).router);
        this.use('/api/client', new ClientApi(config, services).router);

        this.use(
            '/api/frontend',
            conditionalMiddleware(
                () => config.flagResolver.isEnabled('embedProxy'),
                new ProxyController(config, services).router,
            ),
        );

        this.use('/edge', new EdgeController(config, services).router);
    }
}

export default IndexRouter;

module.exports = IndexRouter;
