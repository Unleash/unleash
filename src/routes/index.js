'use strict';

const AdminApi = require('./admin-api');
const ClientApi = require('./client-api');
const FeatureController = require('./client-api/feature.js');

const Controller = require('./controller');
const HealthCheckController = require('./health-check');
const BackstageCTR = require('./backstage.js');
const LogoutController = require('./logout');
const api = require('./api-def');

class IndexRouter extends Controller {
    constructor(config, services) {
        super();
        this.use('/health', new HealthCheckController(config).router);
        this.use('/internal-backstage', new BackstageCTR(config).router);
        this.use('/logout', new LogoutController(config).router);
        this.get(api.uri, this.index);
        this.use(api.links.admin.uri, new AdminApi(config, services).router);
        this.use(api.links.client.uri, new ClientApi(config, services).router);

        // legacy support (remove in 4.x)
        if (config.enableLegacyRoutes) {
            const featureController = new FeatureController(
                services,
                config.getLogger,
            );
            this.use('/api/features', featureController.router);
        }
    }

    index(req, res) {
        res.json(api);
    }
}

module.exports = IndexRouter;
