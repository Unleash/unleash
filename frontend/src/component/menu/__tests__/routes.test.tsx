import { baseRoutes, getRoute } from '../routes';

test('returns all baseRoutes', () => {
    expect(baseRoutes).toMatchSnapshot();
});

test('getRoute() returns named route', () => {
    const featuresRoute = getRoute('/features');
    expect(featuresRoute?.path).toEqual('/features');
});
