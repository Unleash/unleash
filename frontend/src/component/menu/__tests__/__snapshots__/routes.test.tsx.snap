// Vitest Snapshot v1

exports[`returns all baseRoutes 1`] = `
[
  {
    "component": [Function],
    "menu": {},
    "path": "/splash/:splashId",
    "title": "Unleash",
    "type": "protected",
  },
  {
    "component": [Function],
    "enterprise": true,
    "menu": {},
    "parent": "/projects",
    "path": "/projects/create",
    "title": "Create",
    "type": "protected",
  },
  {
    "component": [Function],
    "enterprise": true,
    "menu": {},
    "parent": "/projects",
    "path": "/projects/:projectId/edit",
    "title": ":projectId",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/archive",
    "path": "/projects/:projectId/archived",
    "title": ":projectId",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/projects/:projectId/features/:featureId/:activeTab",
    "path": "/projects/:projectId/features/:featureId/:activeTab/copy",
    "title": "Copy",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/projects",
    "path": "/projects/:projectId/features/:featureId/edit",
    "title": "Edit feature",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/projects",
    "path": "/projects/:projectId/features/:featureId/*",
    "title": "FeatureView",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/projects/:projectId/features",
    "path": "/projects/:projectId/create-toggle",
    "title": "Create feature toggle",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/features",
    "path": "/projects/:projectId/features2/:featureId",
    "title": ":featureId",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "P",
    "menu": {},
    "parent": "/projects",
    "path": "/projects/:projectId/*",
    "title": ":projectId",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "mobile": true,
    },
    "path": "/projects",
    "title": "Projects",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/features",
    "path": "/features/:activeTab/:featureId",
    "title": ":featureId",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "mobile": true,
    },
    "path": "/features",
    "title": "Feature toggles",
    "type": "protected",
  },
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "hidden": false,
    "menu": {
      "mobile": true,
    },
    "path": "/playground",
    "title": "Playground",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/applications",
    "path": "/applications/:name",
    "title": ":name",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "advanced": true,
      "mobile": true,
    },
    "path": "/applications",
    "title": "Applications",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/context",
    "path": "/context/create",
    "title": "Create",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/context",
    "path": "/context/edit/:name",
    "title": ":name",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "advanced": true,
      "mobile": true,
    },
    "path": "/context",
    "title": "Context fields",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/strategies",
    "path": "/strategies/create",
    "title": "Create",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/strategies",
    "path": "/strategies/:name/edit",
    "title": ":name",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/strategies",
    "path": "/strategies/:name",
    "title": ":name",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "advanced": true,
      "mobile": true,
    },
    "path": "/strategies",
    "title": "Strategy types",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/environments",
    "path": "/environments/create",
    "title": "Environments",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "path": "/environments/:id",
    "title": "Edit",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "EEA",
    "menu": {
      "advanced": true,
      "mobile": true,
    },
    "path": "/environments",
    "title": "Environments",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/tag-types",
    "path": "/tag-types/create",
    "title": "Create",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/tag-types",
    "path": "/tag-types/edit/:name",
    "title": ":name",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "advanced": true,
      "mobile": true,
    },
    "path": "/tag-types",
    "title": "Tag types",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/addons",
    "path": "/addons/create/:providerId",
    "title": "Create",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/addons",
    "path": "/addons/edit/:addonId",
    "title": "Edit",
    "type": "protected",
  },
  {
    "component": [Function],
    "hidden": false,
    "menu": {
      "advanced": true,
      "mobile": true,
    },
    "path": "/addons",
    "title": "Addons",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "SE",
    "hidden": false,
    "layout": "main",
    "menu": {},
    "path": "/segments/create",
    "title": "Segments",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "SE",
    "hidden": false,
    "layout": "main",
    "menu": {},
    "path": "/segments/edit/:segmentId",
    "title": "Segments",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "SE",
    "hidden": false,
    "menu": {
      "advanced": true,
      "mobile": true,
    },
    "path": "/segments",
    "title": "Segments",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "adminSettings": true,
    },
    "path": "/history",
    "title": "Event log",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "path": "/archive",
    "title": "Archived toggles",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/admin",
    "path": "/admin/api/create-token",
    "title": "API access",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "RE",
    "menu": {},
    "path": "/admin/create-project-role",
    "title": "Create",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "RE",
    "menu": {},
    "path": "/admin/roles/:id/edit",
    "title": "Edit",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "advanced": true,
      "mobile": true,
    },
    "parent": "/admin",
    "path": "/admin/api",
    "title": "API access",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "adminSettings": true,
    },
    "parent": "/admin",
    "path": "/admin/users",
    "title": "Users",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/admin",
    "path": "/admin/create-user",
    "title": "Users",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/admin",
    "path": "/admin/invite-link",
    "title": "Invite link",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "UG",
    "menu": {
      "adminSettings": true,
    },
    "parent": "/admin",
    "path": "/admin/groups",
    "title": "Groups",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "UG",
    "menu": {},
    "parent": "/admin",
    "path": "/admin/groups/:groupId",
    "title": ":groupId",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "UG",
    "menu": {},
    "parent": "/admin/groups",
    "path": "/admin/groups/create-group",
    "title": "Create group",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "UG",
    "menu": {},
    "parent": "/admin/groups",
    "path": "/admin/groups/:groupId/edit",
    "title": "Edit group",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "RE",
    "menu": {
      "adminSettings": true,
    },
    "parent": "/admin",
    "path": "/admin/roles",
    "title": "Project roles",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "adminSettings": true,
    },
    "parent": "/admin",
    "path": "/admin/auth",
    "title": "Single sign-on",
    "type": "protected",
  },
  {
    "component": [Function],
    "flag": "embedProxyFrontend",
    "menu": {
      "adminSettings": true,
    },
    "parent": "/admin",
    "path": "/admin/cors",
    "title": "CORS origins",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "parent": "/admin",
    "path": "/admin/billing",
    "title": "Billing",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {
      "adminSettings": true,
      "isEnterprise": true,
    },
    "parent": "/admin",
    "path": "/admin-invoices",
    "title": "Billing & invoices",
    "type": "protected",
  },
  {
    "component": [Function],
    "hidden": false,
    "menu": {},
    "path": "/admin",
    "title": "Admin",
    "type": "protected",
  },
  {
    "component": [Function],
    "menu": {},
    "path": "/profile/*",
    "title": "Profile",
    "type": "protected",
  },
]
`;
